src/lib/Sympa/WWW/Marc/Search.pm

```diff
@@ -304,6 +304,11 @@ sub search {
                 last if (/-->/);
             }
         }
+
+        # If $subj is undefined, <$fh> will be undefined thus going further
+        # is useless
+        next unless defined $subj;
```

Comment:
I think it will miss messages matching with `$body` but not subject.

Reply:
If `$subj` is defined before the `while` loop: nothing changes.

If `$subj` is not defined before the `while` loop and is still not defined after that, `<$fh>` is no more defined, since it’s not a `last` that ended the `while` loop but the end of file.

`$lines` of line 326 will always be `''` since the `while` loop of line 327 will never start as `<$fh>` is not defined.

Don’t you think?

Reply:
Up

Reply:
I was confused. There look no problem.